### ISSUE TITLE
Quote comments and suspend feature

### ISSUE THREAD

**Issue by dvidic** [opensourcepos/opensourcepos#1819]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [] Reporting an issue of an unmodified OSPOS installation
- [] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 3.2.0
- OSPOS git commit hash is: 
- PHP version is:      7.1
- MySQL or MariaDB version is:      (e.g. MySQL 5.5, MySQL 5.6, MySQL 5.7, MariaDB 10.0, MariaDB 10.1, MariaDB 10.2)
- OS and version is:      (e.g. CentOS 6.9, Ubuntu 16.4, Windows 10)
- WebServer is:      (e.g. Apache 2.2, Apache 2.4, Nginx 1.12, Nginx 1.13)
- (If applicable) Docker installation: 
- (If applicable) Installation package for the LAMP/LEMP stack is:      (e.g. WAMP, XAMPP)

### Question / Feature

I noticed that Comments written under quotes/invoice don't show up in the actual quote/invoice in print format, only in reports... When I write a comments in quote such as "customers wants items in red color", I need this in actual printed quote. How to fix this? 

Also, why is not possible to suspend quote? I usually talk to customers when making a quote, and its negotiable, and when I go back to quote, it's erased and I have to go all over again....

TNX!

**Comment by odiea** [comment]
Quotes are automatically suspended unless you Discard them.

**Comment by SteveIreland** [comment]
The comments should be appearing in the printed quote.   I'll confirm that's happening, if not then I'll fix it.

**Comment by dvidic** [comment]
I checked once again, comment is missing from quote

**Comment by SteveIreland** [comment]
Of course.  I discovered that I had changed the define name to be more precise but missed a couple of places where the original was still being used.  The fix for that is in the missing quote comment fix for which a pull request is about to be submitted.

**Comment by daN4cat** [comment]
Thanks for the fix, @SteveIreland

**Comment by daN4cat** [comment]
@dvidic could you please test the latest master and confirm  this issue is fixed?

**Comment by dvidic** [comment]
Will do later this evening! TNX!

**Comment by daN4cat** [comment]
Based on the comment here: https://github.com/opensourcepos/opensourcepos/commit/9caf22aea43a5dc6c3c37a79b915fd49fd6c14b2

the comment is fixed.

@SteveIreland there is another issue stated here related to the quote not being hold? I'm not sure you can suspend it or not. You are the one implemented this and therefore better positioned to comment.

Thanks again for the help.

**Comment by odiea** [comment]
@daN4cat . I have not had any issues with the Quote function.
1. Are they selecting a Customer? If not this feature will not work.
2. Are they just selecting the quote from the drop down menu with no Customer?
3. It does automatically suspend the Quote unless the discard button is selected.

**Comment by odiea** [comment]
If there Quote does not appear to be similar then they are not using it correctly. And they must click on the Quote button for it to work properly.

![2018-02-12 13_07_44-wedidit _ powered by ospos 3 2 0](https://user-images.githubusercontent.com/10962177/36117172-eb6e938c-0ff5-11e8-9e84-10f6bfea1a6f.png)

**Comment by daN4cat** [comment]
OK, so @dvidic please check what @odiea is saying so we can close this issue. Thanks.

**Comment by odiea** [comment]
@daN4cat . I cannot do anything with the css but how about replacing optional with Required for Invoicing and Quotes? May clear up some issues.

![2018-02-12 14_07_23-wedidit _ powered by ospos 3 2 0](https://user-images.githubusercontent.com/10962177/36119831-7988c5fe-0ffe-11e8-9bb7-ea50edb944b0.png)
 I currently have it with a new lang line

**Comment by SteveIreland** [comment]
@odiea I will submit a pull request for that tonight unless someone objects.

**Comment by odiea** [comment]
Here is another option. Includes all features. I have no objection. Don't forget to add the land to en-gb for @daN4cat 

![2018-02-12 14_29_46-wedidit _ powered by ospos 3 2 0](https://user-images.githubusercontent.com/10962177/36120714-48ede890-1001-11e8-95f2-8012b0666e95.png)

**Comment by SteveIreland** [comment]
will do

**Comment by dvidic** [comment]
Quote comment appears, but if default invoice comment and/or return policy exist in Configuration it does not work properly...
It can only show one or another (comment or default comment/return policy). I guess default comment is deleted when comment is written under quote, but why return policy dissapears? It would be great to have all that on quote/invoice...

**Comment by SteveIreland** [comment]
I deliberately didn't include the return policy in the quote ... it didn't make sense to me since nothing can be returned yet because nothing has been sold (and actually I deliberately left the comment out in the original design just in case someone wanted to be able to make a note on the sale that didn't appear on the quote).   But that's was just a tentative decision.  I can include the return policy on the quote too if that would be helpful ... just let me know.

My wife had other plans for me tonight so I didn't have a chance to work on anything tonight, so I'm recommitting for tomorrow night.

**Comment by dvidic** [comment]
It would be great to have default footer message for quote/receipt/invoice. I agree that return policy don't have to be on quote, but default comment (such as payment info) should be there plus additional comment made for that particular quote (such as shipping details)...
Thank You!

**Comment by SteveIreland** [comment]
I've made the changes, but there is now enough changes that I should test it before pushing it.  I'll test it tomorrow.

I've added the "Customer is required" message just below the customer selection box.  It will appear ONLY for invoices, quotes and work orders.   If it's a standard sales receipt then, of course, the customer is not required so the message will not appear.

The default invoice comment should now appear on the quote automatically if there isn't a sales specific comment entered.  It's either one or the other, but not both.   I'll test that tomorrow to make sure that is what is happening.

Is the default footer message something new that you are asking for?  What would be the purpose of that and what would the text contain.  Should that be what the default invoice comment is for?  If so, maybe the default invoice comment shouldn't be a substitute for an empty sales comment?

Thoughts and opinions?

**Comment by SteveIreland** [comment]
Today was Valentine's Day so .... well, you know.

And I'm still waiting on opinions about if a default invoice message should be treated as a separate message instead of an "in place of a missing sale comment".  In other words, treating the default invoice message as a "default footer message" for the document.

Also, if I do this for a quote, should it also be done for the invoice?  I don't think it would be applicable to a work order.

So I'm looking at Thursday night instead

**Comment by SteveIreland** [comment]
I didn't get any thoughts on separating out the comments so I did.  Now the default invoice comment set in the configuration, if entered, will appear below the other comments.  The sales comment is right above it.

Also, if the customer is required for the current register mode then "Customer (Required)" will be displayed above the customer field, otherwise "Customer (Optional)" will be displayed.

The pull request has been submitted for these changes.

**Comment by daN4cat** [comment]
@SteveIreland can you please check the review comments so I can merge this to master? Not sure you would need a rebase or not, please have a look.

**Comment by SteveIreland** [comment]
I made the change discussed in this thread.  If someone had an issue with what I was proposing to do then it should have been raised here ... not after I made the change.  

I would argue that the default comment still prints on the invoice but is no longer overridden by the sale comment,.

**Comment by daN4cat** [comment]
Ok

**Comment by daN4cat** [comment]
In master now

**Comment by dvidic** [comment]
@SteveIreland , @daN4cat thank you both, guys, it works now. 
I think that only thing that needs to be done about comments is to have separare default invoice comment, and default quote comment. Quote comments needs to have different text, such as payment info.... If there's a separate invoice/quote comment, this would be perfect and would satisfy everyone, believe me.

**Comment by SteveIreland** [comment]
There isn't going to be any "satisfying everyone".  The goal here is to provide something that most of the OSPOS users can work with.

I can easily enough add a default comment that only appears on quotes and continue to retain the default invoice comment that only appears on the invoice document.  These two comments would always print on their respective document.  These would provide the "footer" comment behavior that you requested.  The comment entered at the point of sale will print on both documents if it is provided.

Is that what you're asking?

**Comment by dvidic** [comment]
Yes, that would be it...

Default comment, if entered, would appear below the other comments like it is now after latest commit, but different for quote and invoice. Those two documents are financially different, that's why there's a need for different default comments.

**Comment by SteveIreland** [comment]
Okay, I can do that this week.  @daN4cat @odiea are you okay with this change?  Anything else to add to it?

**Comment by odiea** [comment]
we made changes to the quote and invoice email because of issues. So if you need to see what was done check the latest master. we did not do anything with the work order.

**Comment by SteveIreland** [comment]
Good ... thanks for the fix.  I haven't had a situation requiring emailing any of the documents so I'm sure that was an oversight on my part.

Since a work order represents an internal operational document I can't imagine when that would be required.

**Comment by daN4cat** [comment]
@SteveIreland please check work order email anyway, just in case.

**Comment by dvidic** [comment]
@SteveIreland Have You found the time to make the changes? 
Thank You

**Comment by SteveIreland** [comment]
I've been working on it as time permits.  A pull request will be submitted for it within the next couple of days.

**Comment by dvidic** [comment]
OK, thank you for your effort! 
I noticed today that user comment does not show on receipt...

**Comment by daN4cat** [comment]
No comments don't go to receipts, comments are for internal use. And please don't change that!

**Comment by daN4cat** [comment]
@SteveIreland's fixes are now in master. Please test and confirm we can close now.

Please note that receipts comments will stay as they are, not printed.

**Comment by dvidic** [comment]
All comments are working as they should. 
@SteveIreland THANK YOU!
